test/network_policy_test.go

```diff
@@ -62,12 +64,271 @@ var _ = Describe(
+				func() {
+					aeroCluster, err := getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.AerospikeNetworkPolicy.TLSAlternateAccessType = asdbv1beta1.AerospikeNetworkTypeOthers
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+				},
+			)
+
+			It(
+				"UpdatingFabricTypeInNetworkPolicy: should fail when fabric type is changed",
+				func() {
+					aeroCluster, err := getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.AerospikeNetworkPolicy.FabricType = asdbv1beta1.AerospikeNetworkTypeOthers
+					aeroCluster.Spec.AerospikeNetworkPolicy.CustomFabricNetworkNames = []string{"ipvlan-conf-1"}
+					aeroCluster.Spec.PodSpec.AerospikeObjectMeta.Annotations = map[string]string{
+						"k8s.v1.cni.cncf.io/networks": "ipvlan-conf-1",
+					}
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+				},
+			)
+
+			It(
+				"UpdatingTLSFabricTypeInNetworkPolicy: should fail when TLS fabric type is changed",
+				func() {
+					aeroCluster, err := getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.AerospikeNetworkPolicy.TLSFabricType = asdbv1beta1.AerospikeNetworkTypeOthers
+					aeroCluster.Spec.AerospikeNetworkPolicy.CustomTLSFabricNetworkNames = []string{"ipvlan-conf-1"}
+					aeroCluster.Spec.PodSpec.AerospikeObjectMeta.Annotations = map[string]string{
+						"k8s.v1.cni.cncf.io/networks": "ipvlan-conf-1",
+					}
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+				},
+			)
+
+			// Added test-case for only 'customAccessNetworkNames`, rest of the types will be similar to this only
+			It(
+				"MissingNetworkNameInPodAnnotations: should fail when access is set to 'others' and "+
+					"customAccessNetworkNames is not present in pod annotations",
+				func() {
+					aeroCluster, err := getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.AerospikeNetworkPolicy.AccessType = asdbv1beta1.AerospikeNetworkTypeOthers
+					aeroCluster.Spec.AerospikeNetworkPolicy.CustomAccessNetworkNames = []string{"ipvlan-conf-1"}
+
+					// define different networks than the ones defined in CustomAccessNetworkNames
+					aeroCluster.Spec.PodSpec.AerospikeObjectMeta.Annotations = map[string]string{
+						"k8s.v1.cni.cncf.io/networks": "ipvlan-conf-2, ipvlan-conf-3",
+					}
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+				},
+			)
+		},
+	)
+}
+
 func doTestNetworkPolicy(
 	multiPodPerHost bool, enableTLS bool, ctx goctx.Context,
 ) {
+	var aeroCluster *asdbv1beta1.AerospikeCluster
+
+	AfterEach(func() {
+		err := deleteCluster(k8sClient, ctx, aeroCluster)
```

Comment:
Actually, we don't need to get the cluster to delete it, just name and namespace is sufficient.
We might have to fix this behaviour at other places of AfterEach to  avoid unnecessary `getCluster` call.
However, we can add a safe check of notFound error to avoid AfterEach failure